code/game/objects/items/potions.dm

```diff
@@ -239,18 +239,20 @@
 	return ishuman(user)
 
 /obj/item/potion/zombie/imbibe_effect(mob/living/carbon/human/user)
-	user.become_zombie_after_death = 2
+	user.become_zombie = TRUE
 
 /obj/item/potion/zombie/impact_atom(atom/target)
 	var/mob/M = get_last_player_touched()
 	var/list/L = get_all_mobs_in_dview(get_turf(src))
 	for(var/mob/living/carbon/human/H in L)
 		if(H.isDeadorDying())
-			if(isjusthuman(H))
+			if(ishuman(H))
 				H.make_zombie(M)
 			else
 				new /mob/living/simple_animal/hostile/necro/skeleton(get_turf(H), M, H.mind)
```

Comment:
you don't need this if/else statement at all, you're ONLY looping through humans, so this will always be true

Reply:
I was just fixing a bug, but I guess I can go through all the code and fix it